bin/reth/src/db/diff.rs

```diff
@@ -0,0 +1,399 @@
+                    find_diffs::<BlockWithdrawals>(primary_tx, secondary_tx, output_dir)?
+                }
+                Tables::TransactionBlock => {
+                    find_diffs::<TransactionBlock>(primary_tx, secondary_tx, output_dir)?
+                }
+                Tables::Transactions => {
+                    find_diffs::<Transactions>(primary_tx, secondary_tx, output_dir)?
+                }
+                Tables::TxHashNumber => {
+                    find_diffs::<TxHashNumber>(primary_tx, secondary_tx, output_dir)?
+                }
+                Tables::Receipts => find_diffs::<Receipts>(primary_tx, secondary_tx, output_dir)?,
+                Tables::PlainAccountState => {
+                    find_diffs::<PlainAccountState>(primary_tx, secondary_tx, output_dir)?
+                }
+                Tables::PlainStorageState => {
+                    find_diffs::<PlainStorageState>(primary_tx, secondary_tx, output_dir)?
+                }
+                Tables::Bytecodes => find_diffs::<Bytecodes>(primary_tx, secondary_tx, output_dir)?,
+                Tables::AccountHistory => {
+                    find_diffs::<AccountHistory>(primary_tx, secondary_tx, output_dir)?
+                }
+                Tables::StorageHistory => {
+                    find_diffs::<StorageHistory>(primary_tx, secondary_tx, output_dir)?
+                }
+                Tables::AccountChangeSet => {
+                    find_diffs::<AccountChangeSet>(primary_tx, secondary_tx, output_dir)?
+                }
+                Tables::StorageChangeSet => {
+                    find_diffs::<StorageChangeSet>(primary_tx, secondary_tx, output_dir)?
+                }
+                Tables::HashedAccount => {
+                    find_diffs::<HashedAccount>(primary_tx, secondary_tx, output_dir)?
+                }
+                Tables::HashedStorage => {
+                    find_diffs::<HashedStorage>(primary_tx, secondary_tx, output_dir)?
+                }
+                Tables::AccountsTrie => {
+                    find_diffs::<AccountsTrie>(primary_tx, secondary_tx, output_dir)?
+                }
+                Tables::StoragesTrie => {
+                    find_diffs::<StoragesTrie>(primary_tx, secondary_tx, output_dir)?
+                }
+                Tables::TxSenders => find_diffs::<TxSenders>(primary_tx, secondary_tx, output_dir)?,
+                Tables::SyncStage => find_diffs::<SyncStage>(primary_tx, secondary_tx, output_dir)?,
+                Tables::SyncStageProgress => {
+                    find_diffs::<SyncStageProgress>(primary_tx, secondary_tx, output_dir)?
+                }
+                Tables::PruneCheckpoints => {
+                    find_diffs::<PruneCheckpoints>(primary_tx, secondary_tx, output_dir)?
+                }
+            };
+        }
+
+        Ok(())
+    }
+}
+
+/// Find diffs for a table, then analyzing the result
+fn find_diffs<'a, T: Table>(
```

Comment:
yeah, because `cursor_read` is generic over T, I'm not sure this can be simplified much further, even in code size, especially since we do sort of want to do this for every possible table